build.gradle.kts

```diff
@@ -52,6 +52,12 @@ val compileJavaJvmArgs by extra((project.properties["compileJavaJvmArgs"] as Str
 val shouldRunAllCheckers by extra(isCI || project.hasProperty("runAllCheckers"))
 
 tasks.register<UpdateEapBuildNumber>("updateEapBuildNumber")
+tasks.register<UpdateIntellijStableVersions>("updateIntellijStableVersions")
+
+tasks.register("updateIntellijVersions") {
+  dependsOn("updateIntellijStableVersions", "updateEapBuildNumber")
+  tasks.getByName("updateEapBuildNumber").mustRunAfter("updateIntellijStableVersions")
```

Comment:
Hmmm I've run `updateIntellijVersions` on the current commit (`788aab04445bf258d5e34ab0baf1726a68bae268`) and ended up with:

```properties
#Mon Aug 08 14:21:34 CEST 2022
latestMinorsOfOldSupportedMajors=2020.3.4,2021.1.3,2021.2.4,2021.3.3,2022.1.4
eapOfLatestSupportedMajor=222.3739.24-EAP-SNAPSHOT
earliestSupportedMajor=2020.3
latestStable=2022.2
```

Here we've got both an EAP and a stable version for 2022.2, which makes little sense. EAPs basically only serve us to check whether we'll be compatible with the new major version once it's stable-released... once major version is stable-released we can just build against the stable version, not EAP :/

Reply:
Shouldn't this be not allowed by the val buildNumberThreshold in UpdateEapBuildNumber?

Reply:
> this

By _this_, you mean the above update to properties file? 🤔

Reply:
This update seems to be the consistent result for `updateIntellijVersions` (i.e. not a race condition or sth), I've run the task a couple times 🤔

Reply:
I get it now, buildNumberThreshold is read from IntellijVersions, which is read from the intellijVersions.properties file but only at the beginning of the program execution. So after the task updateIntellijStableVersions finishes, the values in IntellijVersions are outdated and shouldn't be read.

Reply:
Should we update the values in IntellijVersions object after an update to the properties file happens to avoid any strange behaviour later in the same program execution?

Reply:
> Should we update the values in IntellijVersions object after an update to the properties file happens to avoid any strange behaviour later in the same program execution?

In the spirit of Functional Programming... I'd rather **limit side effects** (like saving stuff to file) to as narrow parts of the code as possible, and instead rely on **pure (deterministic && side-effect-free) functions**  wherever possible.

So the general outline would be:
1. read properties and HTML
2. do **all** processing => deterministic & side-effect-free => always the same result for the same inputs, can be reliably tested without any mocking
3. write the results => side-effect